src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationCodeFixProvider.cs

```diff
@@ -84,33 +91,66 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
 
             if (singleDeclarator)
             {
-                // This was a local statement with a single variable in it.  Just Remove 
-                // the entire local declaration statement.  Note that comments belonging to
+                // This was a local statement with a single variable in it.  Just Remove
+                // the entire local declaration statement. Note that comments belonging to
                 // this local statement will be moved to be above the statement containing
-                // the out-var. 
+                // the out-var.
                 var localDeclarationStatement = (LocalDeclarationStatementSyntax)declaration.Parent;
                 var block = (BlockSyntax)localDeclarationStatement.Parent;
                 var declarationIndex = block.Statements.IndexOf(localDeclarationStatement);
 
-                if (declarationIndex > 0 &&
-                    sourceText.AreOnSameLine(block.Statements[declarationIndex - 1].GetLastToken(), localDeclarationStatement.GetFirstToken()))
+                // Try to find a predecessor Statement on the same line that isn't going to be removed
+                StatementSyntax priorStatementSyntax = null;
+                var localDeclarationToken = localDeclarationStatement.GetFirstToken();
+                for (int i = declarationIndex - 1; i >= 0; i--)
+                {
+                    var statementSyntax = block.Statements[i];
+                    if (declarationsToRemove.Contains(statementSyntax))
+                    {
+                        continue;
+                    }
+
+                    if (sourceText.AreOnSameLine(statementSyntax.GetLastToken(), localDeclarationToken))
+                    {
+                        priorStatementSyntax = statementSyntax;
+                    }
+
+                    break;
+                }
+
+                if (priorStatementSyntax != null)
                 {
                     // There's another statement on the same line as this declaration statement.
                     // i.e.   int a; int b;
                     //
                     // Just move all trivia from our statement to be trailing trivia of the previous
                     // statement
                     editor.ReplaceNode(
```

Comment:
where would other trivia that is in between will go except the one that you explicitly attach? can you add some test that shows how comments are handled?

Reply:
The comment handling is already covered in detail in the non fixall case here https://github.com/dotnet/roslyn/blob/9305ede18b0432055929ef70a2dfdd895a085707/src/EditorFeatures/CSharpTest/InlineDeclaration/CSharpInlineDeclarationTests.cs#L649-L924